src/util/util.c

```diff
@@ -521,3 +521,58 @@ unsigned char *util_zlib_deflate(unsigned char *data, size_t data_in_size, size_
 	buffer = realloc(buffer, *data_out_size);
 	return buffer;
 }
+
+money32 string_to_money(char * string_to_monetise)
+{
+	const char* decimal_char = language_get_string(STR_LOCALE_DECIMAL_POINT);
+	char * text_ptr = string_to_monetise;
+	int i, j;
+	//Remove everything except numbers and decimal
+	for (i = 0; text_ptr[i] != '\0'; ++i) {
+		while (!(
+			(text_ptr[i] >= '0' && text_ptr[i] <= '9') ||
+			(text_ptr[i] == decimal_char[0]) ||
+			(text_ptr[i] == '\0')
+		)) {
+			//move everything over to the left by one
+			for (j = i; text_ptr[j] != '\0'; ++j) {
+				text_ptr[j] = text_ptr[j + 1];
+			}
+			text_ptr[j] = '\0';
+		}
+	}
+	
+	//determine if decimal exists in text
+	char *decimal_place = strstr(string_to_monetise, decimal_char);
+	if (decimal_place == NULL) {
+		//if decimal char does not exist, keep it basic. multiply by 10 to fill decimal and be done.
+		return atoi(string_to_monetise) * 10;
```

Comment:
I think it's the difference between display and internal representation. E.g.: £ 10.00 = 100.

If that's the case, it does not take other exchange rates into account. This will have to be changed.